File: ts/src/test/Exchange/test.fetchTradingFees.ts
import { Exchange } from "../../../ccxt";
import testTradingFee from './base/test.tradingFee.js';
import testSharedMethods from './base/test.sharedMethods.js';

async function testFetchTradingFees (exchange: Exchange, skippedProperties: object) {
    const method = 'fetchTradingFees';
    const fees = await exchange.fetchTradingFees ();
    const symbols = Object.keys (fees);
    testSharedMethods.assertNonEmtpyArray (exchange, skippedProperties, method, symbols);
    for (let i = 0; i < symbols.length; i++) {
        const symbol = symbols[i];
        testTradingFee (exchange, skippedProperties, method, symbol, fees[symbol]);
    }
}

export default testFetchTradingFees;

File: julia/Ccxt/test/exchange/test.fetchTradingFees.jl
using Test
using Ccxt
function testFetchTradingFees(exchange, skippedProperties)

    method = "fetchTradingFees";
    fees = Base.fetch(fetchTradingFees(exchange));
    symbols = objectKeys(fees);
    assertNonEmtpyArray(testSharedMethods, exchange, skippedProperties, method, symbols);
    i = 0
    while functions.ccxtruthy(functions.ccxt_lt(i, length(symbols)))
        symbol = get(symbols, i + 1, nothing);
        testTradingFee(exchange, skippedProperties, method, symbol, get(fees, Symbol(symbol), nothing));
        i += 1
    end
    return true
end
